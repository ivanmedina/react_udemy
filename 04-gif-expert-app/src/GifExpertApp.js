import React, { useState } from 'react';
import ReactDOM from 'react-dom';
import AddCategory from './components/addCategory';
import { GifGrid } from './components/GifGrid';
import './index.css'

export const GifExpertApp=()=>{
    const [categories, setCategories]=useState(['360 flip']);

    // const handleAdd=()=>{
    //     setCategories([...categories,'Skateboarding']);
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            < AddCategory setCategories={setCategories}/>
            <hr/>
            {/* <button onClick={handleAdd}>Agregar</button> */}
            <ol>
                {
                    categories.map(category=>(
                        <GifGrid
                            key={ category }
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
    );
}