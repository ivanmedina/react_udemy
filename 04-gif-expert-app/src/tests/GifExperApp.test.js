import { GifExpertApp } from "../GifExpertApp";
import { shallow } from "enzyme";
describe('Prueba en <GifExpertApp /',()=>{
    
    test('debe de mostrarse correctamente',()=>{
        const wrapper =shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar una lista de categorias',()=>{
        const categories = ['hola','adios'];
        const wraper =shallow(<GifExpertApp defaultCategories={categories}/>);
        
        expect(wraper).toMatchSnapshot();
        expect(wraper.find('GifGrid').length).toBe(categories.length);
        
    })

})