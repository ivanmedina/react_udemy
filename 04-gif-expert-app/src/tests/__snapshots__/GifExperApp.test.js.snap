// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifExpertApp / debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="hola"
      key="hola"
    />
    <GifGrid
      category="adios"
      key="adios"
    />
  </ol>
</Fragment>
`;

exports[`Prueba en <GifExpertApp / debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
